src/mlpack/core/util/rand.hpp

```diff
@@ -0,0 +1,134 @@
+/**
+ * @file core/util/rand.hpp
+ * @author Marcus Edel
+ * @author Omar Shrit
+ *
+ * A simple `randX` wrapper that based on the data type and the function type,
+ * e.g., randi, randn, randu forwards to coot::randX` or `arma::randX`. This
+ * file is initially copied from ensmallen
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_UTIL_RAND_HPP
```

Comment:
There is already a lot of random number support in `math/` (I just documented it).  Although these functions are internal, and we shouldn't document them externally, it would still be a good idea to see if we can make them match our existing RNG functionalities a bit better.

Reply:
maybe mix this file with that one?